src/cross_section/src/cross_section.cpp

```diff
@@ -304,6 +304,129 @@ std::shared_ptr<const PathImpl> CrossSection::GetPaths() const {
   return paths_;
 }
 
+/**
+ * Constructs a CrossSection from a vector of polygons.
+ * No checking is done, so don't have self-intersecting paths
+ * Polygon contours should be wound CCW.
+ * Polygon holes should be wound CW.
+ *
+ * @param polygons A vector of a vector of path points.
+ */
+CrossSection CrossSection::CreateFromPolygonsUnchecked(
```

Comment:
I understand the idea here, but I'm not convinced. This is mostly because "checked" is not an easily-testable condition that I can trust external developers have done in the same way as the library. Anything related to floating-point geometry has this trouble. I think it leaves open too many foot-guns.

Reply:
From what I know so far, no one really complained about the lack of primitives because they implement their own primitives (from code years ago), and just construct the manifold object from their mesh object.
And our checks are really needed to make sure they don't blow their legs off...

Also, I think most of their concerns are about the performance of boolean operations. Mesh construction performance is rarely the bottleneck AFAIK. Maybe the performance for extruding complex SVG is, but those are really complicated and are not convex.

Reply:
btw, isn't `CreateFromVertsAndTriangles` just the manifold constructor that constructs from a mesh?

Reply:
> I understand the idea here, but I'm not convinced. This is mostly because "checked" is not an easily-testable condition that I can trust external developers have done in the same way as the library. Anything related to floating-point geometry has this trouble. I think it leaves open too many foot-guns.

Why do you need to "trust external developers"? When I use these routines I "shoot myself in the foot" a lot.
The world doesn't end.  Surely, it's my problem if I fail to provide proper input to those routines.

Reply:
@briansturgill Please understand that we don't know anything about you or your history, only what you have written to us on Github. The first thing I remember you saying to me was that you weren't planning to use Manifold because you were going to write your own Mesh Boolean in "a couple of weeks". You also recently blamed us for poor design for what turned out to be your own lack of understanding of the basic underpinnings of computational geometry (winding numbers): #750. None of this gives me much confidence in your opinions. 

I'm being harsher here than usual because I don't feel you're giving my collaborators the respect they deserve. Maybe you're the best developer the world has ever known, but to prove it to me, you need to start by being humble, listening well, asking questions, and leading with evidence instead of opinions.